src/identify/index.js

```diff
@@ -172,8 +181,26 @@ class IdentifyService {
     // Get the observedAddr if there is one
     observedAddr = IdentifyService.getCleanMultiaddr(observedAddr)
 
+    let addresses
+
+    try {
+      const envelope = await Envelope.openAndCertify(signedPeerRecord, PeerRecord.DOMAIN)
+      const peerRecord = await PeerRecord.createFromProtobuf(envelope.payload)
+
+      addresses = peerRecord.multiaddrs
+    } catch (err) {
+      log('received invalid envelope, discard it and fallback to listenAddrs is available')
```

Comment:
While it's not a strict failure we should still log the error.
```suggestion
      log('received invalid envelope, discard it and fallback to listenAddrs is available', err)
```